session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -239,7 +244,15 @@ public void insertTablet(Tablet tablet, boolean sorted)
     try {
       RpcUtils.verifySuccess(client.insertTablet(request));
     } catch (TException e) {
-      throw new IoTDBConnectionException(e);
+      if (reconnect()) {
```

Comment:
The reconnect() method will call the open() method again, would this cause an infinite loop？
If the open fails, I think we can just throw an exception, as next time the insert request came can call the reconnect() method.

Reply:
It seems that the open() doesn't call the reconnect(), there won't be any infinite loops.

Reply:
yeah, I read the code in the wrong place, it indeed does not have the problems.